tutorials/reading-time/scripts/content.js

```diff
@@ -0,0 +1,25 @@
+const article = document.querySelector("article");
+
+// `document.querySelector` may return null if the selector doesn't match anything.
+if (article) {
+  const text = article.textContent;
+  /**
+   * count all groups of non-whitespace characters
+   * /\w+/g only matches the latin alphabet
+   * don't count /\s/ b/c this matches empty strings and leading and trailing spaces
+   */
+  const words = text.matchAll(/[^\s]+/g);
```

Comment:
Suggested rewrite to make it a little more obvious what's happening and why.

```suggestion
  /**
   * Regular expression to find all "words" in a string.
   *
   * Here, a "word" is a sequence of one or more non-whitespace characters in a row. We don't use the
   * regular expression character class "\w" to match against "word characters" because it only
   * matches against the Latin alphabet. Instead, we match against any sequence of characters that
   * *are not* a whitespace characters. See the below link for more information.
   * 
   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
   */
  const wordMatchRegExp = /[^\s]+/g;
  const words = text.matchAll(wordMatchRegExp);
```